webhook/webhook_client_impl.go

```diff
@@ -2,12 +2,36 @@ package webhook
 
 import (
 	"context"
+	"errors"
+	"net/url"
+	"strings"
 
 	"github.com/disgoorg/disgo/discord"
 	"github.com/disgoorg/disgo/rest"
 	"github.com/disgoorg/snowflake/v2"
 )
 
+// ParseURL parses a webhook URL
+func ParseURL(webhook string, opts ...ConfigOpt) (Client, error) {
```

Comment:
considering you already return the client here this makes more sense I think
```suggestion
func NewWithURL(webhook string, opts ...ConfigOpt) (Client, error) {
```